AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/HostedUIProviderInfo.swift

```diff
@@ -28,8 +28,8 @@ extension HostedUIProviderInfo: Codable {
 
     init(from decoder: Decoder) throws {
         let values = try decoder.container(keyedBy: CodingKeys.self)
-        idpIdentifier = try values.decode(String.self, forKey: .idpIdentifier)
-        federationProviderName = try values.decode(String.self, forKey: .federationProviderName)
+        idpIdentifier = try? values.decode(String.self, forKey: .idpIdentifier)
+        federationProviderName = try? values.decode(String.self, forKey: .federationProviderName)
```

Comment:
what is the reason of this change?

Reply:
these values are optional, but decoding fails if we are expecting non-optional value. So to avoid it changed the try to optional.

Reply:
We should use `decodeIfPresent` in that case rather than `decode`.